### PR TITLE
Improved ‘compiler’ example showing parsing of an actual language and use of actual interning

### PR DESCRIPTION
**Pull request by regexident**
This PR aims to resolve https://github.com/salsa-rs/salsa/issues/151.

Comments are more than welcome. 🙂

### CODE REVIEW

**Comment by regexident** [comment]
Any chance to have this reviewed/merged?

File: examples/compiler/compiler.rs [hunk 1]
```diff
@@ -1,6 +1,74 @@
-pub trait CompilerDatabase: salsa::Database {
-    fn interner(&self) -> &Interner;
+use std::sync::Arc;
+
+use crate::{interner::Interner, values::*};
+
+#[salsa::query_group(CompilerDatabase)]
+#[salsa::requires(Interner)]
```
Comment (matklad) [inline examples/compiler/compiler.rs]: There's no need for `requires` clause, because there's `: Interner` "public" inheritance anyway.

Comment (regexident) [reply examples/compiler/compiler.rs]: Fixed.

**Comment by regexident** [comment]
Thanks @matklad!

**Pull request merged**